### PR TITLE
use `NewProposal` event to get proposer

### PR DESCRIPTION
**Pull request by dev-matan-tsuberi**

### CODE REVIEW

**Comment by dev-matan-tsuberi** [comment]
[Use `NewContributionProposal` event to get the `proposer` instead of using the proposal's params](https://trello.com/c/2tj23LdU/458-use-newcontributionproposal-event-to-get-the-proposer-instead-of-using-the-proposals-params)

File: src/actions/arcActions.ts [hunk 1]
```diff
@@ -338,7 +340,7 @@ async function getProposalDetails(daoInstance: Arc.DAO, votingMachineInstance: A
     reputationChange: Util.fromWei(contributionProposal.reputationChange),
     periodLength: Number(contributionProposal.periodLength),
     preBoostedVotePeriodLimit,
-    proposer: proposalDetails.proposer,
```
Comment (tibetsprague) [inline src/actions/arcActions.ts]: onProposalExecuted also updates the proposer, so this needs to happen there too

Comment (dev-matan-tsuberi) [reply src/actions/arcActions.ts]: https://github.com/daostack/alchemy/pull/164/commits/23795c8413e09bfc63ca4f9a36355bc069811917

File: src/actions/arcActions.ts [hunk 2]
```diff
@@ -820,8 +820,8 @@ export function onProposalExecuted(avatarAddress: string, proposalId: string, ex
       proposal.reputationWhenExecuted = reputationWhenExecuted;
       proposal.winningVote = decision;
 
-      const gpProposalDetails = await votingMachineInstance.getProposal(proposalId);
-      proposal.proposer = gpProposalDetails.proposer; // Have to do this because redeem sets proposer to 0, to prevent future redemptions for proposer
+      // Have to do this because redeem sets proposer to 0, to prevent future redemptions for proposer
+      proposal.proposer = (await votingMachineInstance.NewProposal({_proposalId: proposalId}, {fromBlock: 0, toBlock: 'latest'}).get(undefined, -1))[0].args._proposer;
```
Comment (tibetsprague) [inline src/actions/arcActions.ts]: hmm, wait, actually this is tricky. We actually want to know that the proposer is 0 here, so we don't show redemptions for the proposer. Or wait, i guess that was fixed on the Arc side? Did you test this?

Comment (dev-matan-tsuberi) [reply src/actions/arcActions.ts]: alchemy needs the `proposal.proposer` to always be non-zero which is guarenteed to be the case when fetching from the event, `getRedemptions(..)` gets the proposer from the blockchain so it'll be zero after a redeem. I don't understand the issue.

Comment (tibetsprague) [reply src/actions/arcActions.ts]: yeah i hear that, but then seems like you can remove a few places where we check for whether proposer is 0x000...

**Pull request merged**